### PR TITLE
Document::canViewItem(): validate itemtype

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Requests on `document.send.php` with invalid  `$_GET['itemtype']` will trigger an error:

```
[2023-05-23 13:13:53] glpiphplog.CRITICAL:   *** Uncaught Exception Error: Class "this_class_does_not_exist" not found in /var/www/glpi/src/Document.php at line 919
  Backtrace :
  src/Document.php:726                               Document->canViewFileFromItem()
  front/document.send.php:55                         Document->canViewFile()
  public/index.php:73                                require()
```

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: src/Document.php [hunk 1]
```diff
@@ -707,6 +707,11 @@ public function canViewFile(array $options = [])
         $itemtype = $options['itemtype'] ?? null;
         $items_id = $options['items_id'] ?? null;
 
+        // Validate itemtype, as it might come directly from $_GET
+        if ($itemtype !== null && !Toolbox::isCommonDBTM($itemtype)) {
```
Comment (trasher) [inline src/Document.php:711]: Is any class that inherits from `CommonGLPI` and not `CommonDBTM` possible here?

Comment (AdrienClairembault) [reply src/Document.php:711]: I think `itemtype` implies to also have some kind of `items_id`, which would indicate a `CommonDBTM`.
As far as a know, we can't link documents to any `CommonGLPI` existing class.

**Comment by trasher** [comment]
Can you take a look to add a test?

File: src/Document.php (hunk 1 above)
Comment (cedric-anne) [inline src/Document.php:711]: IMHO, just adding a `&& is_a($itemtype, CommonDBTM::class, true)` in following block would probably be more easier to understand:
```diff
        if (
            $itemtype !== null
+           && is_a($itemtype, CommonDBTM::class, true)
            && $items_id !== null
            && $this->canViewFileFromItem($itemtype, $items_id)
        ) {
            return true;
        }
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**